modules/ai/assets/js/gutenberg/index.js

```diff
@@ -93,6 +84,21 @@ import { EditTextWithAi } from './edit-text-with-ai';
 			addAiIndicator( 'featured-image', addGenerateFeaturedImageWithAI );
 			addAiIndicatorToTextBlock( [ 'paragraph', 'heading' ] );
 		} );
+
+		const observer = new MutationObserver( ( mutationsList ) => {
+			for ( const mutation of mutationsList ) {
+				if ( 'childList' === mutation.type ) {
+					if ( document.querySelector( '.editor-post-excerpt' ) ) { // Replace with the actual class or ID of the popup
+						addGenerateExcerptWithAI();
+					}
+				}
+			}
+		} );
+
+		observer.observe( document.body, { childList: true, subtree: true } );
+		window.addEventListener( 'beforeunload', () => {
+			observer.disconnect();
+		} );
```

Comment:
can you please add a short explanation of why this is needed in the new version?

Reply:
The observer.observe(document.body, { childList: true, subtree: true }); line sets up a MutationObserver to monitor the entire document for changes in the DOM, specifically to detect when a popup (e.g., the excerpt editor) is added. This allows us to trigger necessary actions whenever the popup appears.

The window.addEventListener('beforeunload', () => { observer.disconnect(); }); line ensures that the observer is properly disconnected when the user navigates away from the current page. This prevents potential memory leaks and keeps our application efficient.